ROADMAP.md

```diff
@@ -0,0 +1,126 @@
+# Roadmap
+
+Contains subheaders about larger improvement efforts.
+
+## `rules_haskell` support
+
+Context: [`rules_haskell`](https://github.com/tweag/rules_haskell) supports
+building Haskell targets by extending Bazel, openly maintained by Tweag.
+It would be useful if `haskell-indexer` supported generating crossreferences for
+`rules_haskell` out of the box.
+
+### Goals
+
+Provide crossreference (as Kythe entries) for both `rules_haskell`-managed
+and `nixpkgs`-sourced compilations. Latter is used by `rules_haskell` for hackage
+imports, instead of hand-rolled source imports.
+
+Crossreference data should be serveable through Kythe's language server, ready to
+be consumed by emacs's lsp-mode (with modifications if needed).
+
+Bonus goal: based on local edits, either make partial index updates possible, or
+merge local updates with the static backing index at serving time.
+
+Possible future expansions:
+  * xref `proto-lens` generated Haskell protobuf code with proto sources.
+    (https://github.com/google/haskell-indexer/issues/15)
+  * xref Haskell FFI code with C code
+    (https://github.com/google/haskell-indexer/issues/18)
+
+#### Rationale
+
+In the non-Bazel world, one might use
+[haskell-ide-engine](https://github.com/haskell/haskell-ide-engine) as a
+[LSP](https://microsoft.github.io/language-server-protocol/) server over a
+stack/cabal workspace, providing editor support (such as type sig requests,
+code navigation, completion).
+
+For Bazel/rules_haskell, one can either hand-roll the stack/cabal files in
+addition to the BUILD files to run the server. But that is duplicate work, also
+there are scaling concerns on large workspaces.
+
+So one way to get LSP support for rules_haskell is to get the Kythe entries
+using haskell-indexer, and serve them through [Kythe's language
+server](https://github.com/kythe/kythe/tree/master/kythe/go/languageserver).
+
+An alternative would be using GHC's HIE files (expected to land in GHC 8.8) to
+write an [LSIF](https://github.com/Microsoft/language-server-protocol/issues/623) index directly, which could be served by `haskell-ide-engine`
+without much overhead. This approach is not pursued here, but might be revisited
+in the future, especially if LSIF demonstrates a solid handling of cross-project
+/ cross-language references (see https://gist.github.com/robinp/76f9d3d91387da5162f773895d4e1d15).
+
+### Tasks
+
+* Decide how compilations will be extracted/analysed.
+
+  * For quick productivity boost, run analysis directly, without extraction.
+    * `rules_haskell`: maybe just indicate indexing by passing some global flag
+      (if possible with Bazel).  Alternatively use `aspects`.
+    * `nixpkgs`: plumb a `should-index` flag to whatever helper is used to define
```

Comment:
It isn't necessary to modify `nixpkgs` at all to index a package. I have a proof of concept implementation here: https://github.com/mpickering/haskell-indexer-nix

Reply:
Thank you! Will have a look.